akka-javasdk/src/main/scala/akka/javasdk/impl/agent/ToolExecutor.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.impl.agent
+
+import akka.annotation.InternalApi
+import akka.javasdk.impl.agent.FunctionTools.FunctionToolInvoker
+import akka.javasdk.impl.serialization.JsonSerializer
+import akka.runtime.sdk.spi.SpiAgent
+
+import java.util.Optional
+import scala.concurrent.ExecutionContext
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+class ToolExecutor(functionTools: Map[String, FunctionToolInvoker], serializer: JsonSerializer) {
+
+  /**
+   * Executes a tool call command synchronously. We use this method in AgentImpl, and the execution context used is the
+   * one provided by the AgentImpl which is the sdkExecutionContext. The goal is to execute the tool call in the
+   * sdkExecutionContext.
+   */
+  def executeAsync(request: SpiAgent.ToolCallCommand)(implicit
+      ex: ExecutionContext): scala.concurrent.Future[String] = {
+    scala.concurrent.Future {
+      execute(request)
+    }
+  }
+
+  /**
+   * Executes a tool call command synchronously. This method is used in the ToolExecutorSpec - no need to go async
+   * there.
+   */
+  private[javasdk] def execute(request: SpiAgent.ToolCallCommand): String = {
```

Comment:
This method is to make the test non-async. 

Async is used only in the prod code.